### PR TITLE
Initial version of event_stream library

### PR DESCRIPTION
**Pull request by ANorwell**
(Copied from readme)
# EventStream

A minimal library for synchronously publishing and subscribing to events.
## Usage

Create an event subscription:

``` ruby
EventStream.subscribe(:my_event) do |event|
  log("#{event.name}, #{event.description}")
```

Then fire an event:

``` ruby
EventStream.publish(:my_event, :description => "An example event.")
```

An event is just an OpenStruct with a name and a bundle of other attributes, so the use of `:description` here is arbitrary.

Events can be subscribed to by name, as above, but many other ways are supported:

``` ruby
# Subscribe to all events
EventStream.subscribe { |e| ... }

# Subscribe to events with a given name
EventStream.subscribe(:my_event) { |e| ... }

# Subscribe to events based on a name regex
EventStream.subscribe(/my/) { |e| ... }

# Subscribe to events based on their attributes
EventStream.subscribe(:user_id => 1) { |e| ... }

# Subscribe to events based on an arbitrary filter
filter = lambda { |e| e.size > 3}
EventStream.subscribe(filter) { |e| ... }
```
### Event Streams

If you wish, you may create more than one event stream:

``` ruby
stream = EventStream::Stream.new
stream.subscribe(...) { |e| ... }
stream.publish(...)
```
## Installation

Add this line to your application's Gemfile:

``` ruby
gem 'event_stream'
```

And then execute:

```
$ bundle
```

Or install it yourself as:

```
$ gem install event_stream
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+A minimal library for synchronously publishing and subscribing to events.
+
+## Usage
+
+Create an event subscription:
+
+```ruby
+EventStream.subscribe(:my_event) do |event|
+  log("#{event.name}, #{event.description}")
```
Comment (jbodah) [inline README.md:11]: missing the `end`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+EventStream.subscribe(:my_event) { |e| ... }
+
+# Subscribe to events based on a name regex
+EventStream.subscribe(/my/) { |e| ... }
+
+# Subscribe to events based on their attributes
+EventStream.subscribe(:user_id => 1) { |e| ... }
+
+# Subscribe to events based on an arbitrary filter
+filter = lambda { |e| e.size > 3}
```
Comment (jbodah) [inline README.md]: whitespace around `{ ... }`

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+require "bundler/gem_tasks"
```
Comment (jbodah) [inline Rakefile:1]: didn't know this exists, cool!

Comment (ANorwell) [inline Rakefile:1]: This comes from `bundle gem <GEM_NAME>` automatically

**Pull request merged**